### PR TITLE
Add debian package for docker

### PR DESCRIPTION
**Pull request by Qubad786**
This adds a debian package for docker. This stuff is taken from how docker is packaging itself for debian.

### CODE REVIEW

**Comment by themylogin** [comment]
Does it have to be in the moby repo? I see it's currently the only thing that'll be different from the upstream or am I mistaken?

**Comment by Qubad786** [comment]
We have a patch into this repo: https://github.com/truenas/moby/pull/1 – So, debian is not an only change in this fork. Am I missing something?

**Comment by themylogin** [comment]
@Qubad786 that's for 20.10, I don't see it in master

**Comment by Qubad786** [comment]
@themylogin [truenas/20.10](https://github.com/truenas/moby/tree/truenas/20.10) is in-sync with upstream 20.10 (+ having our additional updates). That's the version, we're using right now.

File: debian/rules [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# force packages to be built with xz compression, as Ubuntu 21.10 and up use
+# zstd compression, which is non-standard, and breaks 'dpkg-sig --verify'
+override_dh_builddeb:
+	dh_builddeb -- -Zxz
+
+override_dh_auto_clean:
+
+override_dh_auto_build:
+	# install go 1.16
+	curl https://dl.google.com/go/go1.16.4.linux-amd64.tar.gz -o /tmp/go.tar.gz
```
Comment (sonicaj) [inline debian/rules:22]: Why are we installing golang here and not defining it in build depends ?

**Review by sonicaj — COMMENTED** [review]
Nice work!

File: debian/rules (hunk 1 above)
Comment (Qubad786) [reply debian/rules:22]: bullseye's apt mirror comes with [1.15](https://tracker.debian.org/pkg/golang-defaults) but we need at least 1.16 for docker during the build process (on their side, they're building it inside container which already has this version)

**Review by sonicaj — APPROVED** [review]

**Pull request merged**